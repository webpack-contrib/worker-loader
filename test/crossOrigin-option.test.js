import getPort from 'get-port';

import {
  compile,
  getCompiler,
  getErrors,
  getModuleSource,
  getResultFromBrowser,
  getWarnings,
} from './helpers';

describe('"crossOrigin" option', () => {
  it('should not work by default', async () => {
    const compiler = getCompiler('./basic/entry.js');
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats);

    expect(getModuleSource('./basic/worker.js', stats)).toMatchSnapshot(
      'module'
    );
    expect(stats.compilation.assets['test.worker.js']).toBeDefined();
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with crossOrigin enabled', async () => {
    const port = await getPort();
    const compiler = getCompiler('./basic/entry.js', {
      crossOrigin: `http://localhost:${port}/public-path-static/`,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats, port);
    const moduleSource = getModuleSource('./basic/worker.js', stats);

    expect(moduleSource.indexOf('crossOrigin.js') > 0).toBe(true);
    expect(
      moduleSource.indexOf('__webpack_public_path__ + "test.worker.js"') === -1
    ).toBe(true);
    expect(
      moduleSource.indexOf(
        `"http://localhost:${port}/public-path-static/" + "test.worker.js"`
      ) > 0
    ).toBe(true);
    expect(stats.compilation.assets['test.worker.js']).toBeDefined();
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with crossOrigin enabled and "esModule" with "false" value', async () => {
    const port = await getPort();
    const compiler = getCompiler('./basic/entry.js', {
      crossOrigin: `http://localhost:${port}/public-path-static/`,
      esModule: false,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats, port);
    const moduleSource = getModuleSource('./basic/worker.js', stats);

    expect(moduleSource.indexOf('crossOrigin.js') > 0).toBe(true);
    expect(
      moduleSource.indexOf('__webpack_public_path__ + "test.worker.js"') === -1
    ).toBe(true);
    expect(
      moduleSource.indexOf(
        `"http://localhost:${port}/public-path-static/" + "test.worker.js"`
      ) > 0
    ).toBe(true);
    expect(stats.compilation.assets['test.worker.js']).toBeDefined();
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });

  it('should work with crossOrigin enabled and "esModule" with "true" value', async () => {
    const port = await getPort();
    const compiler = getCompiler('./basic/entry.js', {
      crossOrigin: `http://localhost:${port}/public-path-static/`,
      esModule: true,
    });
    const stats = await compile(compiler);
    const result = await getResultFromBrowser(stats, port);
    const moduleSource = getModuleSource('./basic/worker.js', stats);

    expect(moduleSource.indexOf('crossOrigin.js') > 0).toBe(true);
    expect(
      moduleSource.indexOf('__webpack_public_path__ + "test.worker.js"') === -1
    ).toBe(true);
    expect(
      moduleSource.indexOf(
        `"http://localhost:${port}/public-path-static/" + "test.worker.js"`
      ) > 0
    ).toBe(true);
    expect(stats.compilation.assets['test.worker.js']).toBeDefined();
    expect(result).toMatchSnapshot('result');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
    expect(getErrors(stats)).toMatchSnapshot('errors');
  });
});
