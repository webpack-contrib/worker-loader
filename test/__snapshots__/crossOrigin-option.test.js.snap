// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"crossOrigin" option should not work by default: errors 1`] = `Array []`;

exports[`"crossOrigin" option should not work by default: module 1`] = `
"export default function() {
  return new Worker(__webpack_public_path__ + \\"test.worker.js\\");
}
"
`;

exports[`"crossOrigin" option should not work by default: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"crossOrigin" option should not work by default: warnings 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "false" value: errors 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "false" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "false" value: warnings 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "true" value: errors 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "true" value: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"crossOrigin" option should work with crossOrigin enabled and "esModule" with "true" value: warnings 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled: errors 1`] = `Array []`;

exports[`"crossOrigin" option should work with crossOrigin enabled: result 1`] = `"{\\"postMessage\\":true,\\"onmessage\\":true}"`;

exports[`"crossOrigin" option should work with crossOrigin enabled: warnings 1`] = `Array []`;
